src/lib/protocols/atg.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * atg.c
+ *
+ * Copyright (C) 2024 - ntop.org
+ *
+ * nDPI is free software: you can contribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_ATG
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_int_atg_add_connection(struct ndpi_detection_module_struct
+						*ndpi_struct, struct ndpi_flow_struct *flow) {
+  ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_ATG,
+			     NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
+}
+
+
+static void ndpi_search_atg(struct ndpi_detection_module_struct *ndpi_struct,
+				    struct ndpi_flow_struct *flow) {
+  struct ndpi_packet_struct *packet = &ndpi_struct->packet;
+  
+  NDPI_LOG_DBG(ndpi_struct, "search for ATG\n");
+
+  if(packet->tcp != NULL) {
+    if(packet->payload_packet_len >= 8) {
+      u_int16_t eth_ip_port = ntohs(10001);
```

Comment:
Is the port fixed for ATG?

Reply:
I'm sorry for my carelessness, I've corrected it

Reply:
![4BBCFC3F-BA71-4113-8F8C-ED65C6A80471](https://github.com/user-attachments/assets/3cf26b3f-da5f-4e47-bc98-f5c5f3ab695d)